### PR TITLE
DA-531 add git status check

### PR DESCRIPTION
**Pull request by Seeker1911**

### CODE REVIEW

File: rest-api/tools/deploy_app.sh [hunk 1]
```diff
@@ -35,6 +35,12 @@ then
   usage
 fi
 
+if [[ `git status --porcelain` ]]; then
```
Comment (danqrodney) [inline rest-api/tools/deploy_app.sh]: Just to be extra safe, we may want to repeat this check below, at least before the deploy step (line 160).

That way if any files get modified while the DB or config is being updated, we will bail out before deploying.

File: rest-api/tools/deploy_app.sh [hunk 2]
```diff
@@ -35,6 +35,12 @@ then
...
+  # Changes
+  echo "git status must be clean"
+  EXIT 0
```
Comment (danqrodney) [inline rest-api/tools/deploy_app.sh]: You should probably exit with an error code, not zero.

File: rest-api/tools/deploy_app.sh (hunk 1 above)
Comment (danqrodney) [reply rest-api/tools/deploy_app.sh]: Er, sorry. This needs to run after line 107... up here the right version of the code hasn't been checked out yet.

File: rest-api/tools/deploy_app.sh [hunk 3]
```diff
@@ -100,6 +106,12 @@ fi
 echo "${BOLD}Checking out code...${NONE}"
 git checkout $VERSION
 
+if [[ $(git status --porcelain) ]]; then
```
Comment (danqrodney) [inline rest-api/tools/deploy_app.sh]: Can we add one more just before gcloud app deploy?

Comment (Seeker1911) [reply rest-api/tools/deploy_app.sh]: done

**Pull request merged**